examples/visco_elasticity/spring_kelvin_model.py

```diff
@@ -0,0 +1,148 @@
+                        [self.lam0, 2.0 * self.mu0 + self.lam0, self.lam0, 0.0, 0.0, 0.0],
+                        [self.lam0, self.lam0, 2.0 * self.mu0 + self.lam0, 0.0, 0.0, 0.0],
+                        [0.0, 0.0, 0.0, 2.0 * self.mu0, 0.0, 0.0],
+                        [0.0, 0.0, 0.0, 0.0, 2.0 * self.mu0, 0.0],
+                        [0.0, 0.0, 0.0, 0.0, 0.0, 2.0 * self.mu0],
+                    ]
+                )
+                self.I2[0] = 1.0
+                self.I2[1] = 1.0
+                self.I2[2] = 1.0
+
+            case Constraint.PLANE_STRAIN:
+                self.D_0 = np.array(
+                    [
+                        [2.0 * self.mu0 + self.lam0, self.lam0, self.lam0, 0.0],
+                        [self.lam0, 2.0 * self.mu0 + self.lam0, self.lam0, 0.0],
+                        [self.lam0, self.lam0, 2.0 * self.mu0 + self.lam0, 0.0],
+                        [0.0, 0.0, 0.0, 2.0 * self.mu0],
+                    ]
+                )
+                self.I2[0] = 1.0
+                self.I2[1] = 1.0
+
+            case Constraint.PLANE_STRESS:
+                self.D_0 = (
+                        self.E0
+                        / (1 - self.nu ** 2.0)
+                        * np.array(
+                    [
+                        [1.0, self.nu, 0.0, 0.0],
+                        [self.nu, 1.0, 0.0, 0.0],
+                        [0.0, 0.0, 0.0, 0.0],
+                        [0.0, 0.0, 0.0, (1.0 - self.nu)],
+                    ]
+                )
+                )
+                self.I2[0] = 1.0
+                self.I2[1] = 1.0
+
+            case Constraint.UNIAXIAL_STRESS:
+                self.D_0 = np.array([[self.E0]])
+                self.I2[0] = 1.0
+            case _:
+                msg = "Constraint not implemented"
+                raise NotImplementedError(msg)
+
+    def evaluate(
+        self,
+        del_t: float,
+        grad_del_u: np.ndarray,
+        mandel_stress: np.ndarray,
+        tangent: np.ndarray,
+        history: np.ndarray | dict[str, np.ndarray] | None,
+    ) -> None:
+        assert (
+            grad_del_u.size // (self.geometric_dim**2)
+            == mandel_stress.size // self.stress_strain_dim
+            == tangent.size // (self.stress_strain_dim**2)
+        )
+
+        # reshape gauss point arrays
+        n_gauss = grad_del_u.size // (self.geometric_dim ** 2)
+        mandel_view = mandel_stress.reshape(-1, self.stress_strain_dim)
+
+        strain_increment = strain_from_grad_u(grad_del_u, self.constraint).reshape(-1, self.stress_strain_dim)
+        strain_visco_n = history['strain_visco'].reshape(-1, self.stress_strain_dim)
+        strain_n = history['strain'].reshape(-1, self.stress_strain_dim)
+
+        I2 = np.tile(self.I2, n_gauss).reshape(-1, self.stress_strain_dim)
+        tr_eps = np.sum(strain_increment[:, :self.geometric_dim], axis=1)[:, np.newaxis]
+
+        if del_t == 0:
```

Comment:
Can you explain why we need this condition?

Reply:
I added that to include the computation of the time 0, where the visco strain are zero (see test problem._time = 0 -> solve, problem._time=dt -> solve over time loop). The variable _time of teh solver is the del_t in IncrSmallStrainModel interface which is defined as "time at the end of the increment". So for del_t=0 the end time is 0. Without this condition we get an error if problem._time is not set. If you have a better idea here, let me know

Reply:
From a physical standpoint: What is happening before the visco-part? Or is this just a limitation of the viscosity model? To me, it seems a little weird that we first need to have a purely elastic deformation without viscosity that happens without time passing. But if you see no way around it, you can also merge it;)

Reply:
For t=0 the visco part is zero and the behaviour is pure linear elastic.
I could also just assert that del_t != 0? In that case, it wouldn't be possible to compute the zero time step. 

Maybe we have to think about the definition of the variable problem._time / del_t. If that is really a time step (then a 0 time step make nearly no sence; or a time -> then the incsmallstrainproblem needs to compute the del_t from two times)

I will think about it

Reply:
I deleted the zero condition and used in the test 10e-8 as first time step to verify the elastic step

I added the global time variable to the material law interface.

The solver has now a time and a del_t (default = 1). The global time is updated in the update method.